test/03_reduction/reduction_test.cpp

```diff
@@ -160,9 +176,14 @@ namespace hiptensor
                         stream, (bfloat16_t*)resource->hostC().get(), elementsC);
                     stream << std::endl;
 
-                    stream << "Refenrence elements (" << elementsC << "):\n";
+                    stream << "Tensor D elements (" << elementsD << "):\n";
```

Comment:
Can we factor this function away into a templated helper? Lots of repeated code for each dataType...
```
auto print_array_elements= [](auto type_tag, ...) {
    using DataT = typename type_tag::type;
    stream << "Tensor A elements (" << elementsA << "):\n";
                    hiptensorPrintArrayElements<DataT >(
                        stream, (DataT *)resource->hostA().get(), elementsA);
                    stream << std::endl;;
...
};

template <typename DataT> struct type_tag {using type = DataT;};

// Usage:
print_array_elements(type_tag<bfloat16_t>{}, ...)
```